### ISSUE TITLE
control equality checks or provide imperative return values

### ISSUE THREAD

**Issue by aappddeevv** [facebookexperimental/Recoil#139]
I notice that there is no way to control equality checks on returned values, e.g. from selectors "get" definition.

Is there a way to definitely control equality checking? Something like the comparator in React.memo.

Or are we supposed to return a "const selector value" if the object has not changed? I know I could break up an object into more fine grained parts to potentially address this, but that may un-ergonomic.

**Comment by acutmore** [comment]
Hi @aappddeevv. It looks like the Recoil team do plan to have an API similar to this:

>Internally, we're working on a mechanism to limit updates when the values have not changed. But, we're planning to really clean up that API before publishing it.
https://github.com/facebookexperimental/Recoil/issues/41#issuecomment-629601674

**Comment by aappddeevv** [comment]
Thanks for pointing that out, I did not gleam that it was related to this one.

Equality control is pretty important so as long is it can be explicit I'm happy. The issue is also that with non-javascript flavored transpiled languages object equality becomes even more tricky sometimes.

**Comment by drarmstr** [comment]
Yes, as @acutmore pointed out, we have plans for a proper API to control the propagation of updates.  In the meantime, if you're adventurous and don't mind an evolving API and need to unblock yourself, you can play with what we currently use under the `cacheImplementation` selector options.  These control the equality on inputs, vs outputs as you describe.  I expect the final API for this will be very different and closer to what you're requesting.

**Comment by drarmstr** [comment]
Updated temporary API in #907
